dashboard/app/controllers/activities_controller.rb

```diff
@@ -24,7 +24,7 @@ def milestone
 
     if params[:script_level_id]
       @script_level = ScriptLevel.cache_find(params[:script_level_id].to_i)
-      @level = @script_level.level
+      @level = (params[:level_id] && Level.find(params[:level_id].to_i)) || @script_level.levels[0]
```

Comment:
actual note; do we really want to default to the first level, or do we want to be more clever?

We could move that `select_level` method you defined below into the ScriptLevel model and call it something a bit more verbose like `get_first_active_level`, then just call it here and in the other places we care about getting the right level.